packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -976,7 +991,11 @@ export abstract class Chart extends Observable {
             }
         }
 
-        return legendData;
+        legendManager.update();
+    }
+
+    private setLegendData(legendType: ChartLegendType, legend: ChartLegend) {
+        legend.data = this.series.filter((s) => s.properties.showInLegend).flatMap((s) => s.getLegendData(legendType));
```

Comment:
I was imagining that this would not be happening directly anymore?

Reply:
This is for gradient legend which is unchanged, I only added save restore for the category legend